File: java/0049_GroupAnagrams.java
/*
09DEC2025
https://leetcode.com/problems/group-anagrams/
*/

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> groups = new HashMap<>();

        for (String str : strs) {
            int[] charCount = new int[26];
            for (char c : str.toCharArray()) {
                charCount[c - 'a']++;
            }
            String key = Arrays.toString(charCount);
            groups.computeIfAbsent(key, k -> new ArrayList()).add(str);
        }

        return new ArrayList<>(groups.values());

    }
}

File: csharp/0049_GroupAnagrams.cs
/*
21JAN2026
49. Group Anagrams
https://leetcode.com/problems/group-anagrams/

Given an array of strings strs, group the anagrams
together. You can return the answer in any order.
*/

public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var groups = new Dictionary<string, List<string>>();

        foreach (string str in strs) {
            char[] chars = str.ToCharArray();
            Array.Sort(chars);
            string key = new string(chars);
            if (!groups.ContainsKey(key)) {
                groups[key] = new List<string>();
            }
            groups[key].Add(str);
        }

        return new List<IList<string>>(groups.Values);
    }
}
